File: src/main/java/no/nav/sosialhjelp/soknad/consumer/kodeverk/dto/KodeverkDto.java
//package no.nav.sosialhjelp.soknad.consumer.kodeverk.dto;
//
//import com.fasterxml.jackson.annotation.JsonCreator;
//import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
//import com.fasterxml.jackson.annotation.JsonProperty;
//
//import java.util.List;
//import java.util.Map;
//
//@JsonIgnoreProperties(ignoreUnknown = true)
//public class KodeverkDto {
//
//    private final Map<String, List<BetydningDto>> betydninger;
//
//    @JsonCreator
//    public KodeverkDto(
//            @JsonProperty("betydninger") Map<String, List<BetydningDto>> betydninger
//    ) {
//        this.betydninger = betydninger;
//    }
//
//    public Map<String, List<BetydningDto>> getBetydninger() {
//        return betydninger;
//    }
//}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/kodeverk/dto/KodeverkDto.kt
package no.nav.sosialhjelp.soknad.kodeverk.dto

import java.io.Serializable
import java.time.LocalDate

data class KodeverkDto(
    val betydninger: Map<String, List<BetydningDto>>,
) : Serializable

data class BetydningDto(
    val gyldigFra: LocalDate,
    val gyldigTil: LocalDate,
    val beskrivelser: Map<String, BeskrivelseDto>,
) : Serializable

data class BeskrivelseDto(
    val term: String,
    val tekst: String,
) : Serializable
